--- mix2000-1.0.1/build/Makefile.mix	2003-10-14 07:53:59.000000000 +0000
+++ mix2000-1.0.1/build/Makefile.mix-new	2004-11-24 11:20:21.486838184 +0000
@@ -18,6 +18,6 @@
 	find -follow \( -name $(OBJDIR) \) -exec rm -rf {} \; -prune
 
 install:
-	rm /usr/bin/mix2000
 	strip mix/$(OBJDIR)/mix2000
-	cp mix/$(OBJDIR)/mix2000 /usr/bin
+	cp quicktime/i686/libquicktime.a i686/
+	cp -rf mix/$(OBJDIR)/mix2000 $(DESTDIR)/usr/bin
