--- fnv/Makefile	2003-10-03 22:39:17.000000000 +0200
+++ fnv/Makefile-new	2005-07-13 11:42:00.382682971 +0200
@@ -49,9 +49,10 @@
 
 # where to install things
 #
-DESTBIN= /usr/local/bin
-DESTLIB= /usr/local/lib
-DESTINC= /usr/local/include
+prefix=
+DESTBIN= ${prefix}/usr/bin
+DESTLIB= ${prefix}/usr/lib
+DESTINC= ${prefix}/usr/include
 # NOTE: Lines with WWW in them are removed from the shipped Makefile
 WWW= /web/isthe/chroot/html/chongo/src/fnv
 
@@ -195,28 +196,6 @@
 	${INSTALL} -m 0644 ${LIBS} ${DESTLIB}
 	${RANLIB} ${DESTLIB}/libfnv.a
 	${INSTALL} -m 0644 ${HSRC} ${DESTINC}
-	@# NOTE: Lines with WWW in them are removed from the shipped Makefile
-	-@if [ -d ${WWW} ]; then \
-	    echo "rm -f Makefile.ship";			: WWW; \
-	    rm -f Makefile.ship;			: WWW; \
-	    echo "${EGREP} -v WWW Makefile > Makefile.ship"; 	: WWW; \
-	    ${EGREP} -v WWW Makefile > Makefile.ship; 	: WWW; \
-	    echo "rm -f Makefile.save";			: WWW; \
-	    rm -f Makefile.save;			: WWW; \
-	    echo "ln Makefile Makefile.save";		: WWW; \
-	    ln Makefile Makefile.save;			: WWW; \
-	    echo "cp -f Makefile.ship Makefile";	: WWW; \
-	    cp -f Makefile.ship Makefile;		: WWW; \
-	    echo "${TAR} -cf - ${ALL} | ${GZIP_BIN} --best > fnv_hash.tar.gz";: WWW; \
-	    ${TAR} -cf - ${ALL} | ${GZIP_BIN} --best > fnv_hash.tar.gz;: WWW; \
-	    echo "${INSTALL} -m 0644 fnv_hash.tar.gz ${ALL} ${WWW}"; \
-	    ${INSTALL} -m 0644 fnv_hash.tar.gz ${ALL} ${WWW}; \
-	    echo "mv -f Makefile.save Makefile";	: WWW; \
-	    mv -f Makefile.save Makefile;		: WWW; \
-	    echo "rm -f fnv_hash.tar.gz Makefile.ship";	: WWW; \
-	    rm -f fnv_hash.tar.gz Makefile.ship;	: WWW; \
-	fi;						: WWW
-
 clean:
 	-rm -f have_ulong64 have_ulong64.o ll_tmp ll_tmp2 longlong.h
 	-rm -f ${LIBOBJ}
