diff -pruN pkgutils-5.31.0.orig/pkgmk.in pkgutils-5.31.0/pkgmk.in
--- pkgutils-5.31.0.orig/pkgmk.in	2008-02-12 13:42:27.664960128 +1100
+++ pkgutils-5.31.0/pkgmk.in	2008-02-12 13:45:53.616788408 +1100
@@ -319,9 +319,14 @@ check_footprint() {
 				sed 's/^+/NEW       /g' | \
 				sed 's/^-/MISSING   /g' > $FILE.footprint.diff
 			if [ -s $FILE.footprint.diff ]; then
-				error "Footprint mismatch found:"
-				cat $FILE.footprint.diff >&2
-				BUILD_SUCCESSFUL="no"
+				if [ -n "$(egrep '^MISSING   ' $FILE.footprint.diff)" ]; then
+					error "Footprint mismatch found:"
+					cat $FILE.footprint.diff >&2
+					BUILD_SUCCESSFUL="no"
+				else
+					warning "Footprint mismatch found:"
+					cat $FILE.footprint.diff >&2
+				fi
 			fi
 		else
 			warning "Footprint not found, creating new."
