diff -pruN wine-1.0-rc3.orig/dlls/wininet/internet.h wine-1.0-rc3/dlls/wininet/internet.h
--- wine-1.0-rc3.orig/dlls/wininet/internet.h	2008-06-04 12:45:01.344336731 +1000
+++ wine-1.0-rc3/dlls/wininet/internet.h	2008-06-04 12:47:08.197340913 +1000
@@ -45,6 +45,7 @@
 #undef PKCS7_SIGNER_INFO
 #undef X509_NAME
 #undef X509_CERT_PAIR
+#undef X509_EXTENSIONS
 # include <openssl/ssl.h>
 #undef FAR
 #define FAR do_not_use_this_in_wine
