diff -pruN fail2ban-0.8.4.orig/fail2ban-client fail2ban-0.8.4/fail2ban-client
--- fail2ban-0.8.4.orig/fail2ban-client	2011-06-07 23:44:37.226947615 +1000
+++ fail2ban-0.8.4/fail2ban-client	2011-06-07 23:45:38.646947570 +1000
@@ -142,6 +142,7 @@ class Fail2banClient:
 	def __processCmd(self, cmd, showRet = True):
 		beautifier = Beautifier()
 		for c in cmd:
+			time.sleep(0.1)
 			beautifier.setInputCmd(c)
 			try:
 				client = CSocket(self.__conf["socket"])
