diff -pruN lives-0.9.8.10.orig/Makefile.in lives-0.9.8.10/Makefile.in
--- lives-0.9.8.10.orig/Makefile.in	2008-05-01 21:36:49.222263412 +1000
+++ lives-0.9.8.10/Makefile.in	2008-05-01 21:39:10.379263468 +1000
@@ -728,13 +728,9 @@ uninstall-am: uninstall-docDATA
 
 installdirs: 
 	rm -rf $(datadir)/$(PACKAGE)/*
-	mkinstalldirs
-	$(srcdir)/mkinstalldirs \
-	mkdir -p $(DESTDIR)$(bindir) \
-	mkdir -p $(DESTDIR)$(datadir) \
-	mkdir -p $(DESTDIR)$(docdir) \
-	mkdir -p $(DESTDIR)$(sharedir)/applications/ \
-	mkdir -p $(DESTDIR)$(sharedir)/pixmaps/
+	install -d -m 0755 $(DESTDIR)$(bindir)
+	install -d -m 0755 $(DESTDIR)$(datadir)
+	install -d -m 0755 $(DESTDIR)$(docdir)
 
 install-exec-local:
 	rm -rf $(DESTDIR)$(bindir)/lives
@@ -747,8 +743,8 @@ install-exec-local:
 	cp -f $(srcdir)/build-lives-rfx-plugin-multi $(DESTDIR)$(bindir) && chmod a+rx $(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi
 	cp -f $(srcdir)/default.keymap $(DESTDIR)$(datadir)/$(PACKAGE)/
 	$(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi builtin $(DESTDIR)$(datadir) $(DESTDIR)$(bindir)
-	cp -f $(srcdir)/menu-entries/LiVES.desktop $(DESTDIR)$(sharedir)/applications/
-	cp -f $(srcdir)/menu-entries/lives.xpm $(DESTDIR)$(sharedir)/pixmaps/
+	install -m 0644 -D $(srcdir)/menu-entries/LiVES.desktop $(DESTDIR)$(sharedir)/applications/LiVES.desktop
+	install -m 0644 -D $(srcdir)/menu-entries/lives.xpm $(DESTDIR)$(sharedir)/pixmaps/lives.xpm
 
 	$(top_srcdir)/po/update_with_plugins.sh install $(PACKAGE) $(prefix) $(DESTDIR)
 
