diff -Nru gcc-4.0.3-old/gcc/version.c gcc-4.0.3/gcc/version.c
--- gcc-4.0.3-old/gcc/version.c	2006-03-14 02:13:18.000000000 +0100
+++ gcc-4.0.3/gcc/version.c	2006-03-14 02:14:20.000000000 +0100
@@ -5,7 +5,7 @@
    please modify this string to indicate that, e.g. by putting your
    organization's name in parentheses at the end of the string.  */
 
-const char version_string[] = "4.0.3";
+const char version_string[] = "4.0.3 (CRUX)";
 
 /* This is the location of the online document giving instructions for
    reporting bugs.  If you distribute a modified version of GCC,
