diff -pruN pkgutils-5.30.0.orig/pkgmk.in pkgutils-5.30.0/pkgmk.in
--- pkgutils-5.30.0.orig/pkgmk.in	2007-11-29 14:37:43.285643384 +1100
+++ pkgutils-5.30.0/pkgmk.in	2007-11-29 14:54:16.733918048 +1100
@@ -348,7 +348,7 @@ build_package() {
 		
 		cd $PKG
 		info "Build result:"
-		tar czvvf $TARGET *
+		tar --use-compress-program=lzma -cvvf $TARGET *
 		
 		if [ $? = 0 ]; then
 			BUILD_SUCCESSFUL="yes"
@@ -574,7 +574,7 @@ main() {
 	
 	check_pkgfile
 	
-	TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.gz"
+	TARGET="$PKGMK_PACKAGE_DIR/$name#$version-$release.pkg.tar.lzma"
 	
 	if [ "$PKGMK_CLEAN" = "yes" ]; then
 		clean
