diff -pruN lrzip-0.46.orig/Makefile.in lrzip-0.46/Makefile.in
--- lrzip-0.46.orig/Makefile.in	2010-08-23 00:14:37.439002273 +1000
+++ lrzip-0.46/Makefile.in	2010-08-23 00:17:03.140002273 +1000
@@ -102,9 +102,8 @@ install: all
 	mkdir -p $(DESTDIR)${INSTALL_BIN}
 	${INSTALLCMD} -m 755 lrzip $(DESTDIR)${INSTALL_BIN}
 	${INSTALLCMD} -m 755 lrztar $(DESTDIR)${INSTALL_BIN}
-	${LN_S} -f $(DESTDIR)${INSTALL_BIN}/lrzip $(DESTDIR)${INSTALL_BIN}/lrunzip
-	${LN_S} -f $(DESTDIR)${INSTALL_BIN}/lrztar $(DESTDIR)${INSTALL_BIN}/lrzuntar
-	chmod 755 $(DESTDIR)${INSTALL_BIN}/lrzuntar
+	${LN_S} -f lrzip $(DESTDIR)${INSTALL_BIN}/lrunzip
+	${LN_S} -f lrztar $(DESTDIR)${INSTALL_BIN}/lrzuntar
 	mkdir -p $(DESTDIR)${INSTALL_MAN1}
 	${INSTALLCMD} -m 644 $(MAN1FILES) $(DESTDIR)${INSTALL_MAN1}
 	mkdir -p $(DESTDIR)${INSTALL_MAN5}
