--- mtools-3.9.10/Makefile.in	2004-02-28 20:06:35.000000000 +0100
+++ mtools-3.9.10/Makefile.in-new	2005-04-13 22:41:38.674862208 +0200
@@ -218,7 +218,7 @@
 	cd $(infodir) && rm -f mtools.info*
 
 install:	$(bindir)/mtools @BINFLOPPYD@ install-man install-links \
-		$(bindir)/mkmanifest install-scripts install-info
+		$(bindir)/mkmanifest install-scripts
 
 uninstall:	uninstall-bin uninstall-man uninstall-links \
 		uninstall-scripts
