diff -pruN ipset-2.4.6.orig/kernel/ipt_SET.c ipset-2.4.6/kernel/ipt_SET.c
--- ipset-2.4.6.orig/kernel/ipt_SET.c	2009-01-26 20:51:42.000000000 +1100
+++ ipset-2.4.6/kernel/ipt_SET.c	2009-01-26 20:57:03.000000000 +1100
@@ -173,15 +173,15 @@ checkentry(const struct xt_tgchk_param *
 }
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,17)
-static void destroy(void *targetinfo,
+static void destroy(void *targinfo,
 		    unsigned int targetsize)
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(2,6,19)
 static void destroy(const struct xt_target *target,
-		    void *targetinfo,
+		    void *targinfo,
 		    unsigned int targetsize)
 #elif LINUX_VERSION_CODE < KERNEL_VERSION(2,6,28)
 static void destroy(const struct xt_target *target,
-		    void *targetinfo)
+		    void *targinfo)
 #else /* LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,28) */
 static void destroy(const struct xt_tgdtor_param *par)
 #endif
