diff -aur /var/tmp/fbreader-0.99.4.orig/fbreader/src/formats/doc/OleStorage.cpp fbreader-0.99.4/fbreader/src/formats/doc/OleStorage.cpp
--- /var/tmp/fbreader-0.99.4.orig/fbreader/src/formats/doc/OleStorage.cpp	2012-11-30 06:38:24.000000000 -0600
+++ fbreader-0.99.4/fbreader/src/formats/doc/OleStorage.cpp	2017-10-13 22:33:02.043001076 -0500
@@ -59,8 +59,8 @@
 		clear();
 		return false;
 	}
-	static const char OLE_SIGN[] = {0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1, 0};
-	if (std::strncmp(oleBuf, OLE_SIGN, 8) != 0) {
+	static const unsigned char OLE_SIGN[] = {0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1, 0};
+	if (std::memcmp(oleBuf, (signed char *)OLE_SIGN, 8) != 0) {
 		clear();
 		return false;
 	}
