--- tuxracer-0.61/configure.in
+++ tuxracer-0.61/configure.in
@@ -472,10 +472,6 @@
     AC_TRY_COMPILE( [#include <GL/glx.h>], [ char *foo = (char*) glXGetProcAddressARB; ],
                     glx_h_up_to_date=yes, glx_h_up_to_date=no )
     AC_MSG_RESULT([$glx_h_up_to_date])
-
-    if test "x$glx_h_up_to_date" = "xno" ; then
-        AC_MSG_ERROR([Your copy of glx.h is out of date.  You can get a more recent copy from the latest Mesa distribution (http://mesa3d.sourceforge.net).])
-    fi
 fi
 
 dnl Check for glext.h 
