diff -Nru gcc-4.0.3-old/fixincludes/Makefile.in gcc-4.0.3/fixincludes/Makefile.in
--- gcc-4.0.3-old/fixincludes/Makefile.in	2006-03-14 02:13:34.000000000 +0100
+++ gcc-4.0.3/fixincludes/Makefile.in	2006-03-14 02:17:39.000000000 +0100
@@ -163,9 +163,7 @@
 	cp $(srcdir)/../gcc/aclocal.m4 .
 
 check : all
-	autogen -T $(srcdir)/check.tpl $(srcdir)/inclhack.def
-	$(SHELL) ./check.sh $(srcdir)/tests/base
-	@rm -f ./check.sh
+	echo "No check here.."
 
 install : all
 	-rm -rf $(DESTDIR)$(itoolsdir)
