From: Victor Costan <costan@google.com>
Date: Mon, 1 Apr 2019 17:17:47 -0700
Subject: [PATCH] Align version/soversion CMake setup closer with other
 repositories.

PiperOrigin-RevId: 241432456

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ceb5dd9..d50f6c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -213,6 +213,10 @@ target_include_directories(leveldb
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
+
+set_target_properties(leveldb
+  PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
+
 target_compile_definitions(leveldb
   PRIVATE
     # Used by include/export.h when building shared libraries.
@@ -233,10 +237,6 @@ if(BUILD_SHARED_LIBS)
       # Used by include/export.h.
       LEVELDB_SHARED_LIBRARY
   )
-  set_target_properties(leveldb
-    PROPERTIES VERSION ${PROJECT_VERSION})
-  set_target_properties(leveldb
-    PROPERTIES SOVERSION 1)
 endif(BUILD_SHARED_LIBS)
 
 if(HAVE_CLANG_THREAD_SAFETY)
