diff -ur a/include/mbedtls/dhm.h b/include/mbedtls/dhm.h
--- a/include/mbedtls/dhm.h	2018-02-05 16:58:10.000000000 +0000
+++ b/include/mbedtls/dhm.h	2018-02-10 12:29:25.569742120 +0000
@@ -374,7 +374,7 @@
  * in <em>RFC-5114: Additional Diffie-Hellman Groups for Use with
  * IETF Standards</em>.
  */
-#define MBEDTLS_DHM_RFC5114_MODP_P                              \
+#define MBEDTLS_DHM_RFC5114_MODP_2048_P                              \
     MBEDTLS_DEPRECATED_STRING_CONSTANT(                         \
         "AD107E1E9123A9D0D660FAA79559C51FA20D64E5683B9FD1"      \
         "B54B1597B61D0A75E6FA141DF95A56DBAF9A3C407BA1DF15"      \
