diff -pruN hal-0.5.11.orig/hald/linux/blockdev.c hal-0.5.11/hald/linux/blockdev.c
--- hal-0.5.11.orig/hald/linux/blockdev.c	2008-08-27 08:02:30.885326664 +1000
+++ hal-0.5.11/hald/linux/blockdev.c	2008-08-27 08:07:11.005324983 +1000
@@ -1750,7 +1750,7 @@ udev_get_device_file_for_sysfs_path (con
         char *ret;
         char *u_stdout;
         int u_exit_status;
-        const char *argv[] = {"/usr/bin/udevinfo", "--root", "--query", "name", "--path", NULL, NULL};
+        const char *argv[] = {"/sbin/udevadm", "info", "--root", "--query", "name", "--path", NULL, NULL};
         GError *g_error;
 
         ret = NULL;
diff -pruN hal-0.5.11.orig/hald/linux/coldplug.c hal-0.5.11/hald/linux/coldplug.c
--- hal-0.5.11.orig/hald/linux/coldplug.c	2008-08-27 08:02:30.877328769 +1000
+++ hal-0.5.11/hald/linux/coldplug.c	2008-08-27 08:05:28.634330468 +1000
@@ -149,8 +149,8 @@ udev_info_to_hotplug_event (const UdevIn
 static gboolean
 hal_util_init_sysfs_to_udev_map (void)
 {
-	char *udevdb_export_argv[] = { "/usr/bin/udevinfo", "-e", NULL };
-	char *udevroot_argv[] = { "/usr/bin/udevinfo", "-r", NULL };
+	char *udevdb_export_argv[] = { "/sbin/udevadm", "info", "-e", NULL };
+	char *udevroot_argv[] = { "/sbin/udevadm", "info", "-r", NULL };
 	int udevinfo_exitcode;
 	UdevInfo *info = NULL;
 	char *p;
