diff -pruN ppp-2.4.4.orig/chat/Makefile.linux ppp-2.4.4/chat/Makefile.linux
--- ppp-2.4.4.orig/chat/Makefile.linux	2008-07-31 20:08:40.791139905 +1000
+++ ppp-2.4.4/chat/Makefile.linux	2008-07-31 20:08:57.529141744 +1000
@@ -10,7 +10,6 @@ CDEF3=	-UNO_SLEEP			# Use the usleep fun
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
@@ -18,7 +17,7 @@ INSTALL= install
 all:	chat
 
 chat:	chat.o
-	$(CC) -o chat chat.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 chat.o:	chat.c
 	$(CC) -c $(CFLAGS) -o chat.o chat.c
diff -pruN ppp-2.4.4.orig/pppd/Makefile.linux ppp-2.4.4/pppd/Makefile.linux
--- ppp-2.4.4.orig/pppd/Makefile.linux	2008-07-31 20:08:40.780139655 +1000
+++ ppp-2.4.4/pppd/Makefile.linux	2008-07-31 20:08:57.529141744 +1000
@@ -32,7 +32,7 @@ endif
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS+= -Wall
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
diff -pruN ppp-2.4.4.orig/pppd/plugins/Makefile.linux ppp-2.4.4/pppd/plugins/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/Makefile.linux	2008-07-31 20:08:40.781140518 +1000
+++ ppp-2.4.4/pppd/plugins/Makefile.linux	2008-07-31 20:08:57.529141744 +1000
@@ -1,7 +1,8 @@
-CC	= gcc
-COPTS	= -O2 -g
+# CC	= gcc
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
-LDFLAGS	= -shared
+LDFLAGS_PROG	= $(LDFLAGS)
+LDFLAGS	+= -shared
+LIBS = 
 INSTALL	= install
 
 DESTDIR = $(INSTROOT)@DESTDIR@
@@ -23,7 +24,7 @@ all:	$(PLUGINS)
 	for d in $(SUBDIRS); do $(MAKE) $(MFLAGS) -C $$d all; done
 
 %.so: %.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^ $(LIBS)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../patchlevel.h)
 
diff -pruN ppp-2.4.4.orig/pppd/plugins/pppoatm/Makefile.linux ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/pppoatm/Makefile.linux	2008-07-31 20:08:40.785140154 +1000
+++ ppp-2.4.4/pppd/plugins/pppoatm/Makefile.linux	2008-07-31 20:08:57.529141744 +1000
@@ -1,7 +1,5 @@
-CC	= gcc
-COPTS	= -O2 -g
+# CC	= gcc
 CFLAGS	= $(COPTS) -I../.. -I../../../include -fPIC
-LDFLAGS	= -shared
 INSTALL	= install
 
 #***********************************************************************
@@ -33,7 +31,7 @@ endif
 all: $(PLUGIN)
 
 $(PLUGIN): $(PLUGIN_OBJS)
-	$(CC) $(CFLAGS) -o $@ -shared $^ $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^ $(LIBS)
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
diff -pruN ppp-2.4.4.orig/pppd/plugins/radius/Makefile.linux ppp-2.4.4/pppd/plugins/radius/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/radius/Makefile.linux	2008-07-31 20:08:40.784141093 +1000
+++ ppp-2.4.4/pppd/plugins/radius/Makefile.linux	2008-07-31 20:08:57.530139862 +1000
@@ -12,7 +12,7 @@ VERSION = $(shell awk -F '"' '/VERSION/ 
 INSTALL	= install
 
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I. -I../.. -I../../../include -O2 -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
+CFLAGS=$(COPTS) -I. -I../.. -I../../../include -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
@@ -43,13 +43,13 @@ install: all
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
 radius.so: radius.o libradiusclient.a
-	$(CC) -o radius.so -shared radius.o libradiusclient.a
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 radattr.so: radattr.o
-	$(CC) -o radattr.so -shared radattr.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 radrealms.so: radrealms.o
-	$(CC) -o radrealms.so -shared radrealms.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 CLIENTOBJS = avpair.o buildreq.o config.o dict.o ip_util.o \
 	clientid.o sendserver.o lock.o util.o md5.o
diff -pruN ppp-2.4.4.orig/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.4.orig/pppd/plugins/rp-pppoe/Makefile.linux	2008-07-31 20:08:40.782141462 +1000
+++ ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux	2008-07-31 20:08:57.530139862 +1000
@@ -24,18 +24,17 @@ INSTALL	= install
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 VERSION=3.3
 
-COPTS=-O2 -g
 CFLAGS=$(COPTS) -I../../../include/linux
 all: rp-pppoe.so pppoe-discovery
 
 pppoe-discovery: libplugin.a pppoe-discovery.o
-	$(CC) -o pppoe-discovery pppoe-discovery.o libplugin.a
+	$(CC) $(LDFLAGS_PROG) $(CFLAGS) -o pppoe-discovery pppoe-discovery.o libplugin.a
 
 pppoe-discovery.o: pppoe-discovery.c
 	$(CC) $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o pppoe-discovery.o pppoe-discovery.c
 
 rp-pppoe.so: libplugin.a plugin.o
-	$(CC) -o rp-pppoe.so -shared plugin.o libplugin.a
+	$(CC) $(LDFLAGS) $(CFLAGS) -o rp-pppoe.so plugin.o libplugin.a
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
diff -pruN ppp-2.4.4.orig/pppdump/Makefile.linux ppp-2.4.4/pppdump/Makefile.linux
--- ppp-2.4.4.orig/pppdump/Makefile.linux	2008-07-31 20:08:40.771139053 +1000
+++ ppp-2.4.4/pppdump/Makefile.linux	2008-07-31 20:08:57.530139862 +1000
@@ -2,7 +2,7 @@ DESTDIR = $(INSTROOT)@DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 
-CFLAGS= -O -I../include/net
+CFLAGS=$(COPTS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
@@ -10,7 +10,7 @@ INSTALL= install
 all:	pppdump
 
 pppdump: $(OBJS)
-	$(CC) -o pppdump $(OBJS)
+	$(CC) $(LDFLAGS) -o pppdump $(OBJS)
 
 clean:
 	rm -f pppdump $(OBJS) *~
diff -pruN ppp-2.4.4.orig/pppstats/Makefile.linux ppp-2.4.4/pppstats/Makefile.linux
--- ppp-2.4.4.orig/pppstats/Makefile.linux	2008-07-31 20:08:40.769138899 +1000
+++ ppp-2.4.4/pppstats/Makefile.linux	2008-07-31 20:08:57.530139862 +1000
@@ -10,7 +10,6 @@ PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
 COMPILE_FLAGS = -I../include
 LIBS =
 
@@ -26,7 +25,7 @@ install: pppstats
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
-	$(CC) $(CFLAGS) -o pppstats pppstats.c $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o pppstats pppstats.c $(LIBS)
 
 clean:
 	rm -f pppstats *~ #* core
