diff -Nru cdparanoia-III-alpha9.8.orig/interface/utils.h cdparanoia-III-alpha9.8/interface/utils.h
--- cdparanoia-III-alpha9.8.orig/interface/utils.h	2005-04-10 10:33:57.000000000 +0200
+++ cdparanoia-III-alpha9.8/interface/utils.h	2005-04-10 10:34:13.000000000 +0200
@@ -111,7 +111,7 @@
       d->errorbuf=catstring(d->errorbuf,s);
       break;
     case CDDA_MESSAGE_FORGETIT:
-    default:
+    default: ;
     }
   }
 }
@@ -126,7 +126,7 @@
       d->messagebuf=catstring(d->messagebuf,s);
       break;
     case CDDA_MESSAGE_FORGETIT:
-    default:
+    default: ;
     }
   }
 }
@@ -168,7 +168,7 @@
       }
       break;
     case CDDA_MESSAGE_FORGETIT:
-    default:
+    default: ;
     }
   }
   if(malloced)free(buffer);
@@ -204,7 +204,7 @@
 	}
       break;
     case CDDA_MESSAGE_FORGETIT:
-    default:
+    default: ;
     }
   }
   if(malloced)free(buffer);
