--- fontforge/tottf.c	2008/03/12 01:06:02	1.1
+++ fontforge/tottf.c	2008/03/12 01:06:15
@@ -4723,7 +4723,7 @@ static void dumpcmap(struct alltabs *at,
 	format2  = Needs816Enc(sf,&cjklen,map,&apple2,&applecjklen);
 	format14 = NeedsVariationSequenceTable(sf,&vslen,map);
     } else
-	format12 = format2 = format14 = NULL;
+	format12 = format2 = format14 = apple2 = NULL;
 
     /* Two/Three/Four encoding table pointers, one for ms, one for mac */
     /*  usually one for mac big, just a copy of ms */
