--- pgtcl1.5/Makefile.in	2005-07-05 23:14:55.551191544 +0200
+++ pgtcl1.5/Makefile.in-new	2005-07-05 23:17:20.723122056 +0200
@@ -167,7 +167,7 @@
 # for the BINARIES that you specified above have already been done.
 #========================================================================
 
-all: binaries libraries doc
+all: binaries libraries
 
 #========================================================================
 # The binaries target builds executable programs, Windows .dll's, unix
@@ -186,7 +186,7 @@
 	@echo "        xml2nroff sample.xml > sample.n"
 	@echo "        xml2html sample.xml > sample.html"
 
-install: all install-binaries install-libraries install-doc
+install: all install-binaries install-libraries
 
 install-binaries: binaries install-lib-binaries install-bin-binaries
 
