diff -pruN pinentry-0.7.5.orig/configure pinentry-0.7.5/configure
--- pinentry-0.7.5.orig/configure	2008-05-13 05:42:44.000000000 +0000
+++ pinentry-0.7.5/configure	2008-05-13 05:43:12.000000000 +0000
@@ -10629,7 +10629,7 @@ fi
 
 fi
 
-  if test "$no_x" = yes; then
+  if test "$no_x" = no; then
     { { echo "$as_me:$LINENO: error: Can't find X. Please check your installation and add the correct paths!" >&5
 echo "$as_me: error: Can't find X. Please check your installation and add the correct paths!" >&2;}
    { (exit 1); exit 1; }; }
