diff -pruN pygtk-2.12.1.orig/configure.in pygtk-2.12.1/configure.in
--- pygtk-2.12.1.orig/configure.in	2008-08-17 22:37:14.065143263 +1000
+++ pygtk-2.12.1/configure.in	2008-08-17 22:38:06.755144810 +1000
@@ -100,19 +100,22 @@ AC_ARG_ENABLE(thread,
 AM_CHECK_PYMOD(thread,,,enable_thread=no)
 
 dnl Building documentation
-AC_ARG_ENABLE(docs,
-  AC_HELP_STRING([--enable-docs], [Enable documentation building]),enable_docs=$enableval,
-  enable_docs=no)
-if test "${enable_docs}" != no; then
-  dnl Check for xsltproc
-  AC_PATH_PROG([XSLTPROC], [xsltproc])
-  if test -z "$XSLTPROC"; then
-    enable_docs=no
-  fi
-fi
-
-AM_CONDITIONAL(ENABLE_DOCS, test x$enable_docs != xno)
+AC_MSG_CHECKING(whether to enable documentation building with xsltproc)
+AC_ARG_ENABLE(docs, Enable documentation building,
+  [dnl Check for xsltproc
+  AC_CHECK_PROG(have_xsltproc, xsltproc, [yes], [no])
+  if test "x$have_xsltproc" != xno; then
+    build_docs=false
+	 AC_MSG_RESULT(no)
+  else
+    build_docs=true
+    AC_MSG_RESULT(yes)
+  fi],
+  [build_docs=false
+  AC_MSG_RESULT(disabled)])
+AM_CONDITIONAL(ENABLE_DOCS, test x$build_docs = xtrue)
 
+dnl threading
 AC_MSG_CHECKING(whether to enable threading in pygtk)
 if test "x$enable_thread" != xno; then
   extra_mods=gthread
