--- asmutils-0.18/lib/Makefile	2006-02-18 10:39:33.000000000 +0100
+++ asmutils-0.18/lib/Makefile-new	2006-05-15 00:03:13.000000000 +0200
@@ -70,7 +70,7 @@
 		-o $@-static-fastcall $< $(STUB_OBJS) $(LIBC_A)
 endif
 
-all:	libc libm crypto testc testm
+all:	libc libm crypto testc
 
 libc:	$(LIBC_OBJS) $(STUB_OBJS)
 ifdef BUILD_STATIC
--- asmutils-0.18/MCONFIG	2006-05-15 00:11:45.000000000 +0200
+++ asmutils-0.18/MCONFIG-new	2006-05-15 00:13:15.000000000 +0200
@@ -25,7 +25,7 @@
 # in executable even if it is not present in object file.
 # So, if your ld always adds .bss - you can just forget about it.
 
-KERNEL = 24
+KERNEL = 26
 
 # Optimization method (SIZE/SPEED)
 
@@ -97,7 +97,7 @@
 AS := nasm
 LD := ld
 CC := gcc
-MAKE := gmake
+MAKE := make
 
 ASVER := $(shell $(AS) -v | cut -d ' ' -f 3)
 #LDVER := $(shell $(LD) -v | cut -d ' ' -f 4)
