--- hugs98-Mar2005-patched/Makefile	2005-03-12 00:16:22.000000000 +0100
+++ hugs98-Mar2005-patched/Makefile-new	2005-05-14 14:11:54.651842699 +0200
@@ -38,19 +38,12 @@
 	cd libraries; $(MAKE) all
 	cd docs; $(MAKE) all
 
-# We install the standard libraries and the simple demos.
-# We don't install things which don't work on Unix (e.g. Win32).
-
-install: install_all_but_docs
-	cd docs; $(MAKE) install
-
 # Install everything except documentation, which is installed differently
 # by some packagers (e.g. rpm)
 
-install_all_but_docs: fptools src/Makefile
+install: fptools src/Makefile
 	cd src; $(MAKE) install
 	cd libraries; $(MAKE) install
-	cd demos; $(MAKE) install
 
 clean: clean_root
 	cd src; if test -f Makefile; then $(MAKE) clean; fi
