diff -ru ladspa_sdk-orig/src/makefile ladspa_sdk/src/makefile
--- ladspa_sdk-orig/src/makefile	2007-04-01 09:58:07.000000000 +1000
+++ ladspa_sdk/src/makefile	2007-04-01 10:00:34.000000000 +1000
@@ -5,9 +5,9 @@
 # Change these if you want to install somewhere else. In particularly
 # you may wish to remove the middle "local/" part of each entry.
 
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/local/bin/
+INSTALL_PLUGINS_DIR	=	$(DESTDIR)/usr/lib/ladspa/
+INSTALL_INCLUDE_DIR	=	$(DESTDIR)/usr/include/
+INSTALL_BINARY_DIR	=	$(DESTDIR)/usr/bin/
 
 ###############################################################################
 #
@@ -16,7 +16,7 @@
 
 INCLUDES	=	-I.
 LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
+CFLAGS		+=	$(INCLUDES) -Wall -Werror -fPIC
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
@@ -47,19 +47,20 @@
 # TARGETS
 #
 
-test:	/tmp/test.wav ../snd/noise.wav always
-	@echo ---------------------------------------------
-	@echo First listen to the white noise input signal:
-	@echo ---------------------------------------------
-	play ../snd/noise.wav
-	@echo -------------------------
-	@echo Compare to plugin output.
-	@echo -------------------------
-	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	play /tmp/test.wav
+test:
+# /tmp/test.wav ../snd/noise.wav always
+# 	@echo ---------------------------------------------
+# 	@echo First listen to the white noise input signal:
+# 	@echo ---------------------------------------------
+# 	play ../snd/noise.wav
+# 	@echo -------------------------
+# 	@echo Compare to plugin output.
+# 	@echo -------------------------
+# 	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
+# 	play /tmp/test.wav
 
 install:	targets
-	-mkdirhier	$(INSTALL_PLUGINS_DIR)				\
+	mkdir -p	$(INSTALL_PLUGINS_DIR)				\
 			$(INSTALL_INCLUDE_DIR)				\
 			$(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
@@ -123,4 +124,3 @@
 	tar czf `date '+../backup/ladspa_sdk.%Y%m%d%H%M.tgz'` ladspa_sdk/)
 
 ###############################################################################
-
diff -ru ladspa_sdk-orig/src/plugins/sine.cpp ladspa_sdk/src/plugins/sine.cpp
--- ladspa_sdk-orig/src/plugins/sine.cpp	2007-04-01 09:58:07.000000000 +1000
+++ ladspa_sdk/src/plugins/sine.cpp	2007-04-01 10:04:33.000000000 +1000
@@ -59,6 +59,15 @@
 
 /*****************************************************************************/
 
+LADSPA_Handle instantiateSineOscillator(const LADSPA_Descriptor *, unsigned long SampleRate);
+void connectPortToSineOscillator(LADSPA_Handle Instance, unsigned long Port, LADSPA_Data * DataLocation);
+void activateSineOscillator(void * pvHandle);
+void runSineOscillator_FreqAudio_AmpAudio(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqAudio_AmpCtrl(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqCtrl_AmpAudio(LADSPA_Handle Instance, unsigned long SampleCount);
+void runSineOscillator_FreqCtrl_AmpCtrl(LADSPA_Handle Instance, unsigned long SampleCount);
+void cleanupSineOscillator(void *pvHandle);
+
 class SineOscillator {
 private:
 
